status/reporter_gitlab.go

```diff
@@ -132,6 +143,47 @@ func (r *GitLabReporter) setCommitStatus(report TestReport) error {
 	return nil
 }
 
+// updateStatusInComment will create/update a comment in the MR which creates snapshot
+func (r *GitLabReporter) updateStatusInComment(report TestReport) error {
+	comment, err := FormatComment(report.Summary, report.Text)
+	if err != nil {
+		return fmt.Errorf("failed to generate comment for merge-request %d: %w", r.mergeRequest, err)
+	}
+
+	allNotes, _, err := r.client.Notes.ListMergeRequestNotes(r.projectID, r.mergeRequest, nil)
+	if err != nil {
+		return fmt.Errorf("error while getting all comments for merge-request %d: %w", r.mergeRequest, err)
+	}
+	existingCommentId := r.GetExistingNoteID(allNotes, report.ScenarioName, report.SnapshotName)
+	if existingCommentId == nil {
+		noteOptions := gitlab.CreateMergeRequestNoteOptions{Body: &comment}
+		_, _, err := r.client.Notes.CreateMergeRequestNote(r.projectID, r.mergeRequest, &noteOptions)
+		if err != nil {
+			return fmt.Errorf("error while creating comment for merge-request %d: %w", r.mergeRequest, err)
+		}
+	} else {
+		noteOptions := gitlab.UpdateMergeRequestNoteOptions{Body: &comment}
+		_, _, err := r.client.Notes.UpdateMergeRequestNote(r.projectID, r.mergeRequest, *existingCommentId, &noteOptions)
+		if err != nil {
+			return fmt.Errorf("error while creating comment for merge-request %d: %w", r.mergeRequest, err)
+		}
+	}
+
+	return nil
+}
+
+// GetExistingNoteID returns existing GitLab note for the scenario of ref.
+func (r *GitLabReporter) GetExistingNoteID(notes []*gitlab.Note, scenarioName, snapshotName string) *int {
+	for _, note := range notes {
+		if strings.Contains(note.Body, snapshotName) && strings.Contains(note.Body, scenarioName) {
```

Comment:
This makes me think if we retest one merge request, it will create a new snapshot, then there will be a new comment for this scenario and this new snapshot, should we check the scenario name `strings.Contains(note.Body, scenarioName)` only to make sure the old comment for scenario and old snapshot will be overwritten to have a better user experience?
Github reporter has the same usage in [GetExistingCommentID()](https://github.com/redhat-appstudio/integration-service/blob/main/git/github/github.go#L385) but nobody uses github webhook yet so that it has not shown to users but we might need to fix as well in another ticket.
Others look good to me.

Reply:
that's true, we would create duplicated comments, we were discussing to use component name in checkRuns, should we apply the same here?

Reply:
Yes, that is reasonable, I feel.

Reply:
At this point I'm leaning towards replicating the PaC behavior of simply always adding a new comment tbh.

Reply:
ok, disregard my comment in previous thread :D

Reply:
We decided to keep the logic as-is for now and we can revisit it in a follow-up ticket after we start differentiating tests by components.

Reply:
Created [STONEINTG-858](https://issues.redhat.com/browse/STONEINTG-858), let's resolve this discussion for now.